clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -223,9 +223,25 @@ public void connect(String id, InetSocketAddress address, int sendBufferSize, in
     /**
      * Register the nioSelector with an existing channel
      * Use this on server-side, when a connection is accepted by a different thread but processed by the Selector
-     * Note that we are not checking if the connection id is valid - since the connection already exists
+     * <p>
+     * If a connection already exists with the same connection id, the existing zombie connection is closed.
+     * These disconnections are notified and may be processed by the caller using {@link #disconnected()} before
+     * the next {@link #poll(long)}. Kafka brokers add an incrementing index to the connection id to avoid
+     * connection ids being reused even when remote port is reused, so this scenario is highly unlikely.
```

Comment:
Hmm, is it highly unlikely or basically `impossible`? Maybe instead of having a bunch of code for handling this case, we should make it impossible (if we think an int counter is not large enough, we can use a long one) and throw an exception if it happens?